### PR TITLE
LPS-147785 [CP 2.0] Create logic to get form options from API

### PR DESCRIPTION
**Pull request by joanagrego**
PR to create logic to get form options from API.

**How it was developed?**
It was developed using the API to get form field values.

**What has been affected by the development?**
We had to add more props in Radio component to certificate which info message should appear for the user (if it has permanent licence or not) and if the activation key is available.

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:22]: @joanagrego Things got messed up a little here 🤣. Take a look and remove those lines please

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	);
+
+	const productKeyTypes = productTypes?.map((productType) =>
+		productType
+			.map((licenseKey) =>
+				licenseKey.licenseEntryName.replace('DXP ', '')
+			)
+			.sort()
+	);
+
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:88]: I think you can do this directly:
```suggestion
	const productKeyTypes = generateFormValues?.versions?.map(({types}) =>
		types
			.map((licenseKey) =>
				licenseKey.licenseEntryName.replace(DXP_VARIABLE_HERE, '')
			)
			.sort()
	);

```
Also, declare a variable for simple DXP text

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	const returnSelectedVersionIndex = productVersions
+		?.map((label) => label.label)
+		.indexOf(selectedVersion);
+
+	const selectedProductKey =
+		productVersions &&
+		productVersions[returnSelectedVersionIndex].types.find(
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:95]:
```suggestion
		productVersions[returnSelectedVersionIndex]?.types?.find(
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+			(key) => key.licenseEntryName === `DXP ${selectedKeyType}`
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:96]:
```suggestion
			(key) => key.licenseEntryName === `${DXP_VARIABLE} ${selectedKeyType}`
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+												description={`Key activation available: ${subscriptionterm.provisionedCount} of ${subscriptionterm.quantity}`}
+												hasCustomAlert={
+													wasSelected &&
+													displayAlertType
+												}
+												isActivationKeyAvailable={
+													subscriptionterm.provisionedCount >
+													0
+														? true
+														: false
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:284]: Omit the true false return
```suggestion
													subscriptionterm.provisionedCount >
													0
													
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+												}
+												key={index}
+												label={`${getCurrentEndDate(
+													subscriptionterm.startDate
+												)} - ${getCurrentEndDate(
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:289]: Assign this to a variable

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js [hunk 7]
```diff
@@ -57,6 +59,12 @@ const Layout = () => {
 		return;
 	};
 
+	// eslint-disable-next-line no-console
+	console.log('getCurrentpage', getCurrentPage());
+
+	// eslint-disable-next-line no-console
+	console.log('getCurrentProduct', getCurrentProduct());
+
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:67]: console.log here :p

**Review by iagochaves — CHANGES_REQUESTED** [review]
Good job! 👍🏼

**Comment by jairmedeiros** [comment]
Waiting the fixes to start my review :)

**Comment by jairmedeiros** [comment]
@joanagrego, also let's try to use the labels and Assignees to improve our identification :)

**Review by Lucas2691 — APPROVED** [review]
Good job.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js (hunk 1 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:22]: Ooops! Ok, done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js (hunk 4 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:96]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js (hunk 5 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:284]: Done.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js (hunk 6 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:289]: Done.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js (hunk 7 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/layouts/BaseLayout/index.js:67]: Oops! Done.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js (hunk 6 above)
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:289]: Done!

**Review by iagochaves — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+			.map((licenseKey) =>
+				licenseKey.licenseEntryName.replace(DXP_PRODUCT, '')
+			)
+			.sort()
+	);
+
+	const returnSelectedVersionIndex = productVersions
+		?.map((label) => label.label)
+		.indexOf(selectedVersion);
+
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:78]: For the logic that i made, i need the index here and not the value inside this index.

[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js:78]: Actually, for that we would have to use an if(index !== -1) for checks. I think it's better keep with the map.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/routes/Outlets/ProductsMiddlewareOutlet.js [hunk 9]
```diff
@@ -25,7 +25,7 @@ const ProductsMiddlewareOutlet = () => {
 
 	const hasAccessToCurrentProduct = useMemo(() => {
 		const [pageKey] =
-			Object.entries(PAGE_TYPES).find((page) => page[1] === productId) ||
+			Object.entries(PAGE_TYPES)?.find((page) => page[1] === productId) ||
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/routes/Outlets/ProductsMiddlewareOutlet.js:28]: OK

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by iagochaves** [comment]
ci:forward

**Pull request closed without merging**